### DISCUSSION TITLE
Loops in Langflow?

### DISCUSSION THREAD

**Discussion by ryanpeach** [langflow-ai/langflow#3474] [General]
Can one create a loop in langflow? Like, get this output, validate it, if its invalid send this message back to the llm, do it max 3 times, break otherwise and exit early?

**Comment by codenprogressive** [comment]
Indeed this feature is super needed. Not sure if Langflow supports cyclic graphs.

**Comment by ryanpeach** [comment]
I think it would be good if the features somehow guaranteed stopping, like all loops require a max iterations counter, something that makes it qualitatively easier than coding and better for api design (no one wants a hanging api)

**Comment by ChadTimothy** [comment]
This would be amazing

**Comment by severfire** [comment]
I do agree, sometimes when for example I need to rerun same flow because of wrong json format outputted by vertex model

**Comment by ryanpeach** [comment]
I wonder if we could "unroll" a loop and maintain the current DAG data structure. Just in the UI it would be a loop with a max iteration count, and maybe a "break" and "continue" exit clause

**Comment by codenprogressive** [comment]
Agree, this feature will enable and unlock many use cases.

**Comment by severfire** [comment]
@ogabrielluiz @anovazzi1 @lucaseduoli it would be great feature, what do you think about it?

**Reply by severfire** [reply]
any feedback? :-/

**Comment by CaptainKitK** [comment]
Could you not just build a loop with multiple agents?

**Comment by lucaseduoli** [comment]
You can build a loop with the Listen and Notify components and the conditional component, by now. But the Loop functionality itself is already on the roadmap and will be soon implemented! Thanks for the feedback!

**Reply by ryanpeach** [reply]
Thanks! Where is the roadmap issue for tracking?

**Reply by severfire** [reply]
Thank you! Can't wait for this functionality! :-D

**Reply by zcasanova** [reply]
THere happen to be any examples of using listen/notify,and conditional components to loop?

**Reply by severfire** [reply]
I wonder about roadmap as well :-) is there any official document for roadmap?

**Comment by ogabrielluiz** [comment]
Hey all. 
#3694 is part of the implementation of Loops. 

We should have it in main pretty soon.

**Reply by severfire** [reply]
Thank you!

**Reply by guoguoguilai** [reply]
> Hey all. #3694 is part of the implementation of Loops.
> 
> We should have it in main pretty soon.

Is there any tutorial on it?

**Comment by ChadTimothy** [comment]
Is it implemented?

Kind Regards,

Chad

7 Years Bubble Developer | 3 years Venture Capital Studio / Startup Accelerator

[signature_941910469]

📱 Connect on with me : LinkedIn<https://www.linkedin.com/in/chad-wyatt-550795117/>

📧 Email : ***@***.***<mailto:https://www.linkedin.com/in/chad-wyatt-550795117/>

📆 Let’s chat! Schedule a meeting with me : Calendly<https://calendly.com/chad-wyatt/30-min>

“Creativitiy is Intelligence having fun” – Albert Einstein


From: PageGuo ***@***.***>
Date: Tuesday, 29 October 2024 at 9:08 pm
To: langflow-ai/langflow ***@***.***>
Cc: Chad Timothy ***@***.***>, Comment ***@***.***>
Subject: Re: [langflow-ai/langflow] Loops in Langflow? (Discussion #3474)

Hey all. #3694<https://github.com/langflow-ai/langflow/pull/3694> is part of the implementation of Loops.

We should have it in main pretty soon.

Is there any tutorial on it?

**Comment by wahidur028** [comment]
Any feedback?

**Comment by SnakeO** [comment]
Really looking forward to this feature.

**Comment by Stonedcs** [comment]
is it ok?

**Comment by valerypopoff** [comment]
Why is it closed?

**Comment by JeffVince** [comment]
How's it looking?
